File: VB.Net/FastBootFlashingXiaomi/Consoles.vb
Imports System.ComponentModel
Imports System.IO
Module Consoles
    Public Sub Delay(ByVal dblSecs As Double)
        Now.AddSeconds(0.0000115740740740741)
        Dim dateTime As DateTime = Now.AddSeconds(0.0000115740740740741)
        Dim dateTime1 As DateTime = dateTime.AddSeconds(dblSecs)
        While DateTime.Compare(Now, dateTime1) <= 0
            Application.DoEvents()
        End While
    End Sub
    Public Function Fastboot(cmd As String, worker As BackgroundWorker, ee As DoWorkEventArgs) As String
        Console.WriteLine("Fastboot Command : " & cmd)
        Dim output As String = ""
        Dim fastBootExe As Process = New Process()
        fastBootExe.StartInfo.FileName = Application.StartupPath & "\fastboot.exe"
        fastBootExe.StartInfo.Arguments = $"{cmd}"
        fastBootExe.StartInfo.CreateNoWindow = True
        fastBootExe.StartInfo.UseShellExecute = False
        fastBootExe.StartInfo.RedirectStandardOutput = True
        fastBootExe.StartInfo.RedirectStandardError = True

        If worker.CancellationPending Then
            fastBootExe.Dispose()
            ee.Cancel = True
            Return output
        Else
            fastBootExe.Start()
            Dim readerStdError As StreamReader = fastBootExe.StandardError
            Dim readerStdOutput As StreamReader = fastBootExe.StandardError
            output = readerStdError.ReadToEnd() & readerStdOutput.ReadToEnd()
            fastBootExe.WaitForExit()
        End If
        Console.WriteLine(output)
        Return output

    End Function

End Module

File: C#/FastBootFlashingXiaomi/Consoles.cs
using System;
using System.ComponentModel;
using System.Diagnostics;
using System.Windows.Forms;

namespace FastBootFlashingXiaomi
{
    static class Consoles
    {
        public static void Delay(double dblSecs)
        {
            DateTime.Now.AddSeconds(0.0000115740740740741d);
            var dateTime = DateTime.Now.AddSeconds(0.0000115740740740741d);
            var dateTime1 = dateTime.AddSeconds(dblSecs);
            while (DateTime.Compare(DateTime.Now, dateTime1) <= 0)
                Application.DoEvents();
        }
        public static string Fastboot(string cmd, BackgroundWorker worker, DoWorkEventArgs ee)
        {
            Console.WriteLine("Fastboot Command : " + cmd);
            string output = "";
            var fastBootExe = new Process();
            fastBootExe.StartInfo.FileName = Application.StartupPath + @"\fastboot.exe";
            fastBootExe.StartInfo.Arguments = $"{cmd}";
            fastBootExe.StartInfo.CreateNoWindow = true;
            fastBootExe.StartInfo.UseShellExecute = false;
            fastBootExe.StartInfo.RedirectStandardOutput = true;
            fastBootExe.StartInfo.RedirectStandardError = true;

            if (worker.CancellationPending)
            {
                fastBootExe.Dispose();
                ee.Cancel = true;
                return output;
            }
            else
            {
                fastBootExe.Start();
                var readerStdError = fastBootExe.StandardError;
                var readerStdOutput = fastBootExe.StandardError;
                output = readerStdError.ReadToEnd() + readerStdOutput.ReadToEnd();
                fastBootExe.WaitForExit();
            }
            Console.WriteLine(output);
            return output;

        }

    }
}